src/lib.rs

```diff
@@ -1213,6 +1213,66 @@ impl<T: TensorType + PartialEq> PartialEq for Tensor<T> {
     }
 }
 
+fn write_tensor_recursive<T: Display + Debug>(f: &mut Formatter, shape: &[u64], values: &[T]) -> ::std::fmt::Result {
+    match shape.len() {
+        // Handle special case of a scalar tensor.
+        0 => write!(f, "{}", values[0]),
+        1 => {
```

Comment:
Yeah, I put it in because I felt like it was unnecessary to split the values into a bunch of arrays with only 1 element and then do a function call. But the implementation is much nicer without. I guess the cost is at least one function call per value now, unless the function is inlined by rust.

Reply:
This new version may be less efficient, but it's more maintainable and less likely to contain bugs.  If someone runs into a situation where displaying tensors is a major performance cost, we can revisit this, but I suspect that in real-world situations, the cost of formatting tensors should be negligible compared to the rest of the program.